R/stats-mlm-tidiers.R

```diff
@@ -0,0 +1,99 @@
+#' @templateVar class mlm
+#' @template title_desc_tidy
+#'
+#' @param x An `mlm` object created by [stats::lm()] with a matrix as the
+#'   response.
+#' @template param_confint
+#' @template param_quick
+#' @template param_unused_dots
+#'
+#' @evalRd return_tidy(regression = TRUE)
+#'
+#' @details In contrast to `lm` object (simple linear model), tidy output for
+#'   `mlm` (multiple linear model) objects contain an additional column
+#'   `response`.
+#'
+#'   If you have missing values in your model data, you may need to refit
+#'   the model with `na.action = na.exclude`.
+#'
+#' @examples
+#'
+#' mod <- lm(cbind(mpg, disp) ~ wt, mtcars)
+#' tidy(mod, conf.int = TRUE)
+#' @export
+#' @seealso [tidy()]
+#' @family lm tidiers
+#' @export
+
+tidy.mlm <- function(x,
+                     conf.int = FALSE,
+                     conf.level = .95,
+                     ...) {
+
+  # adding other details from summary object
+  s <- summary(x)
+  ret <- tidy.summary.mlm(s)
+
+  # adding confidence intervals
+  if (isTRUE(conf.int)) {
```

Comment:
```suggestion
  if (conf.int) {
```

Reply:
This criticism is valid when you want to use `all.equal()` but not valid in general. `if(isTRUE(logical_var))` is redundant. We ask the user to supply a logical to `conf.int`, so it's reasonable to operate on the assumption they've done this. If we want to valid that `conf.int` is a logical and not `NA` or `NaN` or `NULL`, we could do that, but we should separate that logical out into distinct input validation infrastructure. At some point, we should add input validation to tidiers. That's currently a low priority, however.